ports/stm32/i2c.c

```diff
@@ -285,8 +285,22 @@ int i2c_write(i2c_t *i2c, const uint8_t *src, size_t len, size_t next_len) {
 
 STATIC uint16_t i2c_timeout_ms[MICROPY_HW_MAX_I2C];
 
+static uint32_t i2c_get_id(i2c_t *i2c) {
+    // Mask the lower 16 bits to get the bus base address
+    uint32_t base = ((uint32_t)i2c) & 0xFFFF0000;
+
+    switch (base) {
+        #if defined(STM32H7)
+        case D3_APB1PERIPH_BASE:
+            return 3;
+        #endif
+        default:
```

Comment:
I suggest making this simpler like this:
```c
#if defined(STM32H7)
if (i2c == I2C4) {
    return 3;
}
#endif

return ((uint32_t)i2c - I2C1_BASE) / (I2C2_BASE - I2C1_BASE);
```

Reply:
Okay will change it to that, but the idea was if more I2C are added to D3, then they can be used to calculate the `i2c_id` same way it's done from D2.

Reply:
It looks like the only I2C5 that's currently available is on D2_APB1PERIPH_BASE

Reply:
I think I saw LPI2Cs in the drivers too, not sure which domain they belong to. I changed it to the above suggestion, using the base pointer, and did the same for the clock enable.

Reply:
Actually I2C3 is not always defined, I reverted the clock enable to using IDs.